src/xtgeo/grid3d/grid_properties.py

```diff
@@ -419,7 +441,7 @@ def props(self, propslist):
         self._consistency_check()
 
     @property
-    def dates(self):
+    def dates(self) -> None | list[str | None]:
```

Comment:
More of a nit, but maybe switch to `list[str | None] | None`. I think it better captures optionality rather than seeming like returning `None` is the expected default (`.. or None`)